app/uk/gov/hmrc/soletraderidentification/repositories/JourneyDataRepository.scala

```diff
@@ -16,120 +16,101 @@
-        JourneyIdKey -> journeyId,
-        AuthInternalIdKey -> authInternalId
-      ),
-      Some(Json.obj(
-        JourneyIdKey -> 0
-      ))
-    ).one[JsObject]
-
-  def updateJourneyData(journeyId: String, dataKey: String, data: JsValue, authInternalId: String): Future[UpdateWriteResult] =
-    collection.update(true).one(
-      Json.obj(
-        JourneyIdKey -> journeyId,
-        AuthInternalIdKey -> authInternalId
+      filterJourneyConfig(journeyId, authInternalId)
+    ).headOption()
+
+  def updateJourneyData(journeyId: String, authInternalId: String, dataKey: String, data: JsValue): Future[Boolean] =
+    collection.updateOne(
+      Filters.and(
+        Filters.equal(JourneyIdKey, journeyId),
+        Filters.equal(AuthInternalIdKey, authInternalId)
       ),
-      Json.obj(
-        "$set" -> Json.obj(dataKey -> data)
-      ),
-      upsert = false,
-      multi = false
-    ).filter(_.n == 1)
+      Updates.set(dataKey, Codecs.toBson(data)),
+      UpdateOptions().upsert(false)
+    ).toFuture.map {
+      _.getMatchedCount == 1
+    }
 
-  def removeJourneyDataField(journeyId: String, authInternalId: String, dataKey: String): Future[UpdateWriteResult] =
-    collection.update(true).one(
-      Json.obj(
-        JourneyIdKey -> journeyId,
-        AuthInternalIdKey -> authInternalId
-      ),
-      Json.obj(
-        "$unset" -> Json.obj(dataKey -> 1)
-      ),
-      upsert = false,
-      multi = false
-    ).filter(_.n == 1)
+  def removeJourneyDataField(journeyId: String, authInternalId: String, dataKey: String): Future[Boolean] =
+    collection.updateOne(
+      filterJourneyConfig(journeyId, authInternalId),
+      Updates.unset(dataKey)
+    ).toFuture.map {
+      _.getMatchedCount == 1
+    }
 
-  def removeJourneyData(journeyId: String, authInternalId: String): Future[UpdateWriteResult] =
-    collection.update(true).one(
-      Json.obj(
-        JourneyIdKey -> journeyId,
-        AuthInternalIdKey -> authInternalId
+  def removeJourneyData(journeyId: String, authInternalId: String): Future[Boolean] =
+    collection.findOneAndReplace(
+      Filters.and(
+        Filters.equal(JourneyIdKey, journeyId),
+        Filters.equal(AuthInternalIdKey, authInternalId)
       ),
       Json.obj(
         JourneyIdKey -> journeyId,
         AuthInternalIdKey -> authInternalId,
         CreationTimestampKey -> Json.obj("$date" -> Instant.now.toEpochMilli)
-      ),
-      upsert = false,
-      multi = false
-    ).filter(_.n == 1)
-
-  private val TtlIndexName = "SoleTraderDataExpires"
-
-  private lazy val ttlIndex = Index(
-    key = Seq((CreationTimestampKey, IndexType.Ascending)),
-    name = Some(TtlIndexName),
-    options = BSONDocument("expireAfterSeconds" -> appConfig.timeToLiveSeconds)
-  )
-
-  private def setIndex(): Unit = {
-    collection.indexesManager.drop(TtlIndexName) onComplete {
-      _ => collection.indexesManager.ensure(ttlIndex)
+      )
+    ).toFuture.map {
+      _ != null
```

Comment:
By default "findOneAndReplace" returns the original document not the replacement. Maybe we should make the method return the replacement. One for the team to discuss maybe.

Reply:
Yeah makes sense to discuss with the team